## Analysis 2
1. How many injuries in this dataset involve a skateboard?
2. Of those injuries, what percentage were male and what percentage were female?
3. What was the average age of someone injured in an incident involving a skateboard?

In [1]:
#Importing required packages
import pandas as pd
import matplotlib.pyplot
import seaborn as sb
import numpy as np

In [2]:
#Reading the required files
df1=pd.read_csv(r"NEISS data/NEISS2014.csv")

In [3]:
#Filtering only the accidents involving skateboard
#From the documentation, product code for skateboard is 1333. Filtering data if prod1 or prod2 is a skateboard
df2=df1[(df1['prod1']==1333)|(df1['prod2']==1333)]
print(df2.head())

     CPSC Case # trmt_date  psu   weight stratum  age     sex  \
2      150144993   8/17/14   63  99.7040       M   21  Female   
39     140505206   4/25/14   63  81.5760       M   14    Male   
166    140851075   5/25/14   14  41.0402       M   14    Male   
292    140735930   7/13/14   90   5.7174       C   15  Female   
320    140735911   7/13/14   90   5.7174       C    5    Male   

                       race race_other  diag diag_other  body_part  \
2               None listed        NaN    57        NaN         83   
39                    White        NaN    57        NaN         33   
166                   White        NaN    64        NaN         37   
292                   White        NaN    62        NaN         75   
320  Black/African American        NaN    53        NaN         76   

     disposition  location  fmv  prod1   prod2  \
2              1         0    0   1333     NaN   
39             1         0    0   1333     NaN   
166            1         9    0   1842

In [4]:
print("Count of injuries involving a skateboard is "+str((df2.shape[0])))

Count of injuries involving a skateboard is 495


##### Of those injuries, what percentage were male and what percentage were female?

In [5]:
#Calculating percentage of male and female involved in skateboard accident
df3=df2.groupby(['sex'])['sex'].agg(['count']).reset_index()
print(df3.head())
df3=df3.rename(columns={'count': 'Frequency'})
print(df3.head())

      sex  count
0  Female     89
1    Male    406
      sex  Frequency
0  Female         89
1    Male        406


In [6]:
df3['Percentage']=round((df3['Frequency']/len(df2))*100,2)

In [7]:
print(df3)

      sex  Frequency  Percentage
0  Female         89       17.98
1    Male        406       82.02


##### Alternate method to check skateboard related injuries

In [8]:
df4=df1[(df1['narrative'].str.contains('SKATEBOARD'))]

In [9]:
print('The number of injuries involving a skateboard by checking the narrative description is '+str((df4.shape[0])))

The number of injuries involving a skateboard by checking the narrative description is 466


In [10]:
#Calculating percentage of male and female involved in skateboard accident
df5=df4.groupby(['sex'])['sex'].agg(['count']).reset_index()
df5=df5.rename(columns={'count': 'Frequency'})
df5['Percentage']=round((df5['Frequency']/len(df4))*100,2)
print(df5)

      sex  Frequency  Percentage
0  Female         82        17.6
1    Male        384        82.4


##### For further analysis, we consider df2 (495 records)

##### What was the average age of someone injured in an incident involving a skateboard?

In [11]:
d = df2['age'].describe()
print(d)

count    495.000000
mean      17.890909
std        9.049807
min        3.000000
25%       12.000000
50%       16.000000
75%       21.000000
max       71.000000
Name: age, dtype: float64


In [12]:
print("Age ranges from "+str(d['min'])+" to "+str(d['max']))

Age ranges from 3.0 to 71.0


In [13]:
#Ages above 115 is coded as 114, lets check if age=114 exists
print('Count of ages over 115 is '+ str(len(np.where(df2['age']==114)[0])))

Count of ages over 115 is 0


In [14]:
#Ages less than 2 years are coded as 200 (2 followed by months) and above
#We can code values greater than 200 as 1 and retain other values 
#Creating a new column for this purpose
df2.loc[df2['age']>=200,'age_year'] = 1
df2.loc[df2['age']<200,'age_year'] = df2['age']

c:\program files (x86)\python\lib\site-packages\pandas\core\indexing.py:337: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[key] = _infer_fill_value(value)
c:\program files (x86)\python\lib\site-packages\pandas\core\indexing.py:517: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [15]:
avgage=round(df2['age_year'].mean(),0)
print('The average age of a person in a skateboard accident is '+str(avgage))

The average age of a person in a skateboard accident is 18.0
